src/main/java/nl/novi/behuiswerkles11/services/TelevisionService.java

```diff
@@ -0,0 +1,53 @@
+package nl.novi.behuiswerkles11.services;
+
+import nl.novi.behuiswerkles11.dtos.TelevisionRequestDto;
+import nl.novi.behuiswerkles11.exceptions.RecordNotFoundException;
+import nl.novi.behuiswerkles11.models.Television;
+import nl.novi.behuiswerkles11.repositories.TelevisionRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TelevisionService {
+
+    private final TelevisionRepository televisionRepository;
+
+    public TelevisionService(TelevisionRepository televisionRepository) {
+        this.televisionRepository = televisionRepository;
+    }
+
+    //    Functie voor het ophalen van alle tv's
+    public List<Television> getTelevisions() {
+        return televisionRepository.findAll();
+    }
+
+    //    Functie voor het ophalen van 1 tv
+    public Television getTelevision(Long id) {
+        return televisionRepository.findById(id).orElseThrow(() -> new RecordNotFoundException("Television not found"));
+    }
+
+    //    POST Functie voor het toevoegen van 1 tv
+    public Television addTelevision(Television television) {
+        televisionRepository.save(television);
+        return television;
+    }
+
+    //    PUT Functie voor het aanpassen van 1 tv
+//    public Television updateTelevision(Long id, Television television) {
+//        Optional<Television> televisionOptional = televisionRepository.findById(id);
+//        if (televisionOptional.isPresent()) {
+//            televisionOptional.get();
+//        }
+//        return television;
```

Comment:
Je returned hier de television die je in de parameter ontvangt zonder enige aanpassing. 
Wat je wilt doen is alle waardes uit de "television" (van de parameter) een-voor-een overzetten naar de television die in de "televisionOptional" (van de repository) zit.
Daarna diezelfde television uit de repository weer saven, maar met aangepaste waardes.

Reply:
Je ID blijft hier nu op null staan, omdat de gereturnde "television" in feite de input DTO is die je in de controller ontvangt en door stuurt naar de service. Jou input DTO's hebben geen id, dus vandaar dat het null is.